### PR TITLE
[firebase_storage] Add macOS support

### PR DESCRIPTION
**Pull request by cbenhagen**
## Description

Adds macOS support. Already includes fix proposed in https://github.com/FirebaseExtended/flutterfire/pull/2099.

/cc @collinjackson @franciscojma86 

## Related Issues

https://github.com/FirebaseExtended/flutterfire/issues/1653

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] If the pull request affects only one plugin, the PR title starts with the name of the plugin in brackets (e.g. [cloud_firestore])
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_storage/example/pubspec.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ dependencies:
     sdk: flutter
   firebase_storage:
     path: ../
-  firebase_core: ^0.4.0
```
Comment (collinjackson) [inline packages/firebase_storage/example/pubspec.yaml]: This will probably fail until the updated firebase_core is published.

Comment (cbenhagen) [reply packages/firebase_storage/example/pubspec.yaml]: 0.4.4+2 is already published. Should be fine.

**Comment by cbenhagen** [comment]
@franciscojma86 // @collinjackson thanks for the quick review. Can you merge this?

**Comment by AndroidNils** [comment]
Any updates?

**Comment by franciscojma86** [comment]
I just fixed the merge conflicts and added the user agent fix. I forgot that the google bot doesn't like co-authored PRs :( @cbenhagen can you respond with the "I consent" message? If that doesn't work, I think you can just squash my commits and force push

**Comment by cbenhagen** [comment]
I consent

**Comment by franciscojma86** [comment]
I think it has to be `@googlebot I consent. `

**Comment by cbenhagen** [comment]
@googlebot I consent

**Comment by franciscojma86** [comment]
@googlebot I consent.

**Comment by cbenhagen** [comment]
Thanks @franciscojma86

**Comment by franciscojma86** [comment]
No worries! Sorry for the delay. Somehow I lost track of this PR :)

**Pull request merged**